main.rb

```diff
@@ -0,0 +1,122 @@
+# Enumerable
+module Enumerable
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_each
+    return to_enum(__method__) unless block_given?
+
+    if is_a?(Array)
+      size.times do |i|
+        yield self[i]
+      end
+    else size.times do |i|
+      yield [keys[i], values[i]]
+    end
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(__method__) unless block_given?
+
+    size.times do |i|
+      yield self[i], i
+    end
+  end
+
+  def my_select
+    return to_enum(__method__) unless block_given?
+
+    array = []
+    my_each do |i|
+      array << i if yield i
+    end
+    array
+  end
+
+  def my_all?(*args)
+    if block_given?
+      my_each { |i| return false unless yield i }
+    elsif args != [] && args[0].class != Regexp
+      my_each { |i| return false unless i.is_a?(args[0]) }
+    elsif args[0].class == Regexp
+      my_each { |i| return false if (args[0] =~ i).nil? }
+    else
+      my_each { |i| return false unless i.nil? }
+    end
+    true
+  end
+
+  def my_any?(*args)
+    if block_given?
+      my_each { |i| return true if yield i }
+    elsif args != [] && args[0].class != Regexp
+      my_each { |i| return true if i.is_a?(args[0]) }
+    elsif args[0].class == Regexp
+      my_each { |i| return true unless (args[0] =~ i).nil? }
+    else
+      my_each { |i| return true if i.nil? }
+    end
+    false
+  end
+
+  def my_none?(*args)
+    if block_given?
+      my_each { |i| return false if yield i }
+    elsif args != [] && args[0].class != Regexp
+      my_each { |i| return false if i.is_a?(args[0]) }
+    elsif args[0].class == Regexp
+      my_each { |i| return false unless (args[0] =~ i).nil? }
+    else
+      my_each { |i| return false if i == true }
+    end
+    true
+  end
+
+  def my_count(*args)
+    @var = self
+    empty_arr = []
+
+    if block_given?
+      my_each { |i| empty_arr << i if yield i }
+
+      empty_arr.length
+    else
+      !args.empty? ? @var.length - (self - args).length : length
+    end
+  end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_count

- [ ]  returns the number of items in enum through enumeration::range
```ruby
(1..5).my_count == (1..5).count => 5
```
- Yours is giving an error
![image](https://user-images.githubusercontent.com/52209067/85831447-91840500-b74b-11ea-957c-41d72efc652e.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/85831480-9c3e9a00-b74b-11ea-90b1-ecf1b2e0cd62.png)